magenta-lib/src/main/scala/magenta/deployment_type/AmiCloudFormationParameter.scala

```diff
@@ -34,21 +38,34 @@ object AmiCloudFormationParameter extends DeploymentType {
       implicit val keyRing = resources.assembleKeyring(target, pkg)
       val reporter = resources.reporter
 
-      val stackName = target.stack.nameOption.filter(_ => prependStackToCloudFormationStackName(pkg, target, reporter))
-      val stageName = Some(target.parameters.stage.name).filter(_ => appendStageToCloudFormationStackName(pkg, target, reporter))
-      val cloudFormationStackNameParts = Seq(stackName, Some(cloudFormationStackName(pkg, target, reporter)), stageName).flatten
-      val fullCloudFormationStackName = cloudFormationStackNameParts.mkString("-")
+      val cloudFormationStackLookupStrategy = {
+        if (cloudformationStackByTags(pkg, target, reporter)) {
+          assert(pkg.apps.size == 1, s"The $name package type can only be used when exactly one app is specified - you have [${pkg.apps.map(_.name).mkString(",")}]")
+          assert(target.stack.nameOption.isDefined, s"The $name package type can only be used when a stack is specified")
+          LookupByTags(Map(
+            "Stage" -> target.parameters.stage.name,
+            "Stack" -> target.stack.nameOption.get,
+            "App" -> pkg.pkgApps.map(_.name).head
```

Comment:
You could zip two options and map over it, but ultimately the assertions have been made above. A TODO comment noting that this could be removed once the legacy format is removed might be of use.